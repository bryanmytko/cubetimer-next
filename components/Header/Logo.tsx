import Link from "next/link";
import Image from "next/image";
import styles from "./Logo.module.css";

const Logo = () => (
  <Link href="/" className="flex items-center">
    <Image alt="logo" className="h-10 p-1" src="/assets/logo.png" />
    <div className="text-xl">cubetimer.io</div>
  </Link>
);

export default Logo;
