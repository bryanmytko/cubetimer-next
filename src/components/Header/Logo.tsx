import Link from "next/link";
import Image from "next/image";

interface LogoProps {
  dark?: boolean;
}

const Logo = (props: LogoProps) => {
  const { dark } = props;

  return (
    <Link href="/" className="flex items-center mt-1 md:mt-0">
      <Image
        height="47"
        width="45"
        alt="logo"
        className="h-10 p-1"
        src="/assets/logo.png"
      />
      <div className={`text-xl ${dark ? "text-black" : "text-white"}`}>
        cubetimer.io
      </div>
    </Link>
  );
};

export default Logo;
