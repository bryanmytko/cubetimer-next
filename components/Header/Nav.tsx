import Link from "next/link";
import { useSession, signIn, signOut } from "next-auth/react";
import { Button, Dropdown, Text, Grid, User } from "@nextui-org/react";
import gravatar from "gravatar";

import Logo from "./Logo";

const Nav = () => {
  const { data: session } = useSession();
  const auth = () => {
    if (session) {
      // @TODO google avatars sometimes 403
      // const avatarUrl = session.user?.image || gravatar.url(session.user?.email || "", { protocol: 'https', s: '100' });
      const avatarUrl = gravatar.url(session.user?.email || "", {
        protocol: "https",
        s: "100",
      });
      const displayEmail = session.user?.email || "";
      const displayName = session.user?.name || "";

      return (
        <Grid.Container justify="flex-start" gap={0}>
          <Grid>
            <Dropdown placement="bottom-left">
              <Dropdown.Trigger>
                <User
                  bordered
                  as="button"
                  size="md"
                  color="gradient"
                  src={avatarUrl}
                  name=""
                >
                  <Text size={14} color="white" css={{ d: "flex" }}>
                    {displayName}
                  </Text>
                  <Text size={12} color="$gray500" css={{ d: "flex" }}></Text>
                </User>
              </Dropdown.Trigger>
              <Dropdown.Menu color="primary" aria-label="User Actions">
                <Dropdown.Item
                  key="user"
                  css={{ height: "$18" }}
                  textValue="user"
                >
                  <Text color="$gray900" css={{ d: "flex" }}>
                    Signed in as
                  </Text>
                  <Text b color="$gray900" css={{ d: "flex" }}>
                    {displayEmail}
                  </Text>
                </Dropdown.Item>
                <Dropdown.Item
                  key="statistics"
                  textValue="statistics"
                  withDivider
                >
                  Statistics
                </Dropdown.Item>
                <Dropdown.Item key="profile" textValue="profile" withDivider>
                  Profile
                </Dropdown.Item>
                <Dropdown.Item key="settings" textValue="settings">
                  Settings
                </Dropdown.Item>
                <Dropdown.Item
                  key="logout"
                  color="error"
                  textValue="logout"
                  withDivider
                >
                  <Link href="#" onClick={() => signOut()}>
                    Log Out
                  </Link>
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </Grid>
        </Grid.Container>
      );
    }

    return (
      <Button
        auto
        color="error"
        css={{
          background: "rgb(250 202 21 / var(--tw-bg-opacity))",
          color: "black",
        }}
        href="#"
        onClick={() => signIn()}
      >
        Login
      </Button>
    );
  };

  return (
    <nav className="px-2 sm:px-4">
      <div className="container flex flex-wrap items-center justify-between mx-auto">
        <Logo />
        {/* <button
          data-collapse-toggle="navbar-default"
          type="button"
          className="inline-flex items-center p-2 ml-3 text-sm bg-gray-700 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:focus:ring-gray-600"
          aria-controls="navbar-default"
          aria-expanded="false"
        >
          <span className="sr-only">Open Navigation</span>
          <svg
            className="w-6 h-6"
            aria-hidden="true"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fillRule="evenodd"
              d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 
              011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 
              110 2H4a1 1 0 01-1-1z"
              clipRule="evenodd"
            ></path>
          </svg>
        </button> */}
        <div className="hidden w-full md:block md:w-auto" id="navbar-default">
          <ul className="flex flex-col p-4 mt-4 border md:flex-row md:space-x-6 md:mt-0 md:text-sm md:font-medium md:border-0">
            <li>
              <Link className="block py-2 text-white" href="/">
                Timer
              </Link>
            </li>
            {/* <li>
              <Link className="block py-2 text-white" href="statistics">
                Statistics
              </Link>
            </li> */}
            {/* <li>
              <Link className="block py-2 text-white" href="profile">
                Profile
              </Link>
            </li> */}
            <li>
              <Link className="block py-2 text-white" href="tutorials">
                Tutorials
              </Link>
            </li>
            <li>
              <Link className="block py-2 text-white" href="reviews">
                Reviews
              </Link>
            </li>
            <li>{auth()}</li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Nav;
