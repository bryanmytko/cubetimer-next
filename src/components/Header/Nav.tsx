import Link from "next/link";
import { useSession, signIn, signOut } from "next-auth/react";
import {
  Button,
  Dropdown,
  DropdownMenu,
  DropdownTrigger,
  DropdownItem,
  User,
} from "@nextui-org/react";
import router, { useRouter } from "next/router";
import gravatar from "gravatar";

import Logo from "./Logo";
import { Key } from "react";

const Nav = () => {
  const { data: session } = useSession();
  const { asPath } = useRouter();

  const navItems = [
    { url: "/", name: "timer" },
    { url: "/tutorials", name: "tutorials" },
    { url: "/reviews", name: "reviews" },
  ];

  const auth = () => {
    if (session) {
      const avatarUrl = gravatar.url(session.user?.email || "", {
        protocol: "https",
        s: "100",
      });
      const displayEmail = session.user?.email || "";
      const displayName = session.user?.name || "";

      const dropdownAction = (key: Key) => {
        if (key === "logout") return signOut();
        return router.push({ pathname: String(key) });
      };

      return (
        <div className="grid justify-start">
          <Dropdown>
            <DropdownTrigger>
              <User as="button" name="" avatarProps={{ src: avatarUrl }}>
                <p>{displayName}</p>
              </User>
            </DropdownTrigger>
            <DropdownMenu
              onAction={dropdownAction}
              color="primary"
              aria-label="User Actions"
            >
              <DropdownItem key="user" textValue="user">
                <p>Signed in as:</p>
                <p className="font-bold">{displayEmail}</p>
              </DropdownItem>
              <DropdownItem key="statistics" textValue="statistics">
                Statistics
              </DropdownItem>
              <DropdownItem key="profile" textValue="profile">
                Profile
              </DropdownItem>
              <DropdownItem key="settings" textValue="settings">
                Settings
              </DropdownItem>
              <DropdownItem key="logout" color="warning" textValue="logout">
                Log Out
              </DropdownItem>
            </DropdownMenu>
          </Dropdown>
        </div>
      );
    }

    return (
      <Button color="warning" onClick={() => signIn()}>
        Login
      </Button>
    );
  };

  return (
    <nav className="px-2">
      <div className="container flex flex-wrap items-center justify-between mx-auto">
        <Logo />
        <div className="block w-auto" id="navbar-default">
          <ul className="flex flex-row px-4 py-3 pt-4 space-x-6 text-sm font-medium border-0">
            {navItems.map((item, index) => {
              return (
                <li
                  key={`nav-item-${index}`}
                  className={`${
                    item.url === asPath ? "text-yellow-300" : "text-white"
                  }`}
                >
                  <Link className="block py-2 capitalize" href={item.url}>
                    {item.name}
                  </Link>
                </li>
              );
            })}
            <li>{auth()}</li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Nav;
