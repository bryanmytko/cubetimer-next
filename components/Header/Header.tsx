import Nav from "./Nav";

const Header = () => {
  return (
    <header className="bg-neutral-900 text-gray-100">
      <Nav />
    </header>
  );
};

export default Header;
