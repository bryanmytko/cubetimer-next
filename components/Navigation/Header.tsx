import Nav from './Nav';

const Header = () => {
  return <header className="bg-amber-500">
    <Nav />
  </header>
};

export default Header;
