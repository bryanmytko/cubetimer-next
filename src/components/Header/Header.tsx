import Nav from "./Nav";

const Header = () => {
  return (
    <header className="bg-cyan-800 text-gray-100">
      <Nav />
    </header>
  );
};

export default Header;
