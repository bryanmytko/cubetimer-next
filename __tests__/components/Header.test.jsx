import { render } from "@testing-library/react";

import "@testing-library/jest-dom";

import Header from "../../src/components/Header/Header";

describe("<Header>", () => {
  it("renders the component", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
