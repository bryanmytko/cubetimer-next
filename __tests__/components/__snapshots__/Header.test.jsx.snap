// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders the component 1`] = `
<div>
  <header
    class="bg-cyan-800 text-gray-100"
  >
    <nav
      class="px-2"
    >
      <div
        class="container flex flex-col md:flex-row flex-wrap items-center justify-around md:justify-between mx-auto"
      >
        <a
          class="flex items-center mt-1 md:mt-0"
          href="/"
        >
          <img
            alt="logo"
            class="h-10 p-1"
            data-nimg="1"
            decoding="async"
            height="47"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Flogo.png&w=96&q=75"
            srcset="/_next/image?url=%2Fassets%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Fassets%2Flogo.png&w=96&q=75 2x"
            style="color: transparent;"
            width="45"
          />
          <div
            class="text-xl text-white"
          >
            cubetimer.io
          </div>
        </a>
        <div
          class="block w-auto"
          id="navbar-default"
        >
          <ul
            class="flex flex-row px-4 pb-2 md:pb-3 pt-0 md:pt-4 space-x-6 text-sm font-medium items-center"
          >
            <li
              class="text-white"
            >
              <a
                class="block py-2 capitalize"
                href="/"
              >
                timer
              </a>
            </li>
            <li
              class="text-white"
            >
              <a
                class="block py-2 capitalize"
                href="/tutorials"
              >
                tutorials
              </a>
            </li>
            <li
              class="text-white"
            >
              <a
                class="block py-2 capitalize"
                href="/reviews"
              >
                reviews
              </a>
            </li>
            <li>
              <div
                class="grid justify-start"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased"
                  data-slot="trigger"
                  id="react-aria-:r0:"
                  type="button"
                >
                  <span
                    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
                    tabindex="-1"
                  >
                    <img
                      alt="avatar"
                      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                      src="https://s.gravatar.com/avatar/d41d8cd98f00b204e9800998ecf8427e?s=100"
                    />
                  </span>
                  <div
                    class="inline-flex flex-col items-start"
                  >
                    <span
                      class="text-small text-inherit"
                    />
                    <span
                      class="text-tiny text-foreground-400"
                    />
                  </div>
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
